apis/python/src/tiledbsoma/_dataframe.py

```diff
@@ -880,7 +880,11 @@ def _canonicalize_schema(
                 f"All index names must be defined in the dataframe schema: '{index_column_name}' not in {schema_names_string}",
             )
         dtype = schema.field(index_column_name).type
-        if not pa.types.is_dictionary(dtype) and dtype not in [
+        if pa.types.is_dictionary(dtype):
+            raise TypeError(
+                f"Cannot set index column '{index_column_name}' to an enumeration. Index columns do not support enumerations."
+            )
+        if dtype not in [
```

Comment:
super tiny nit: the array of types being tested against should be defined in a global constant, as it is a base TileDB characteristic....